src/coreComponents/constitutive/contact/ContactBase.hpp

```diff
@@ -0,0 +1,161 @@
+
+  /// Default move constructor
+  ContactBaseUpdates( ContactBaseUpdates && ) = default;
+
+  /// Deleted default constructor
+  ContactBaseUpdates() = default;
+
+  /// Deleted copy assignment operator
+  ContactBaseUpdates & operator=( ContactBaseUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  ContactBaseUpdates & operator=( ContactBaseUpdates && ) =  delete;
+
+  /**
+   * @brief Evaluate the effective aperture, and its derivative wrt aperture
+   * @param[in] aperture the model aperture/gap
+   * @param[out] dEffectiveAperture_dAperture the derivative of the effective aperture wrt aperture
+   * @return An effective physical aperture that is always > 0
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  virtual real64 computeEffectiveAperture( real64 const aperture,
+                                           real64 & dEffectiveAperture_dAperture ) const
+  { GEOSX_UNUSED_VAR( aperture, dEffectiveAperture_dAperture ); return 0.0; }
+
+
+  /**
+   * @brief Evaluate the traction vector and its derivatives wrt to pressure and jump
+   * @param[in] dispJump the displacement jump
+   * @param[in] tractionVector the traction vector
+   * @param[out] dTractionVector_dJump the derivative of the traction vector wrt jump
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  virtual void computeTraction( arraySlice1d< real64 const > const & dispJump,
+                                arraySlice1d< real64 > const & tractionVector,
+                                arraySlice2d< real64 > const & dTractionVector_dJump ) const
+  { GEOSX_UNUSED_VAR( dispJump, tractionVector, dTractionVector_dJump ); }
+
+  /**
+   * @brief Update the traction with the pressure term
+   * @param[in] pressure the pressure term
+   * @param[in] isOpen a flag specifying whether the fracture is open or closed
+   * @param[inout] traction the current tractionVector
+   * @param[out] dTraction_dPressure the derivative of the fist component of traction wrt pressure
+   * @return the updated traction
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  virtual void addPressureToTraction( real64 const & pressure,
+                                      bool const isOpen,
+                                      arraySlice1d< real64 >const & tractionVector,
+                                      real64 & dTraction_dPressure ) const
+  { GEOSX_UNUSED_VAR( pressure, isOpen, tractionVector, dTraction_dPressure ); }
+
+  /**
+   * @brief Evaluate the limit tangential traction norm and return the derivative wrt normal traction
+   * @param[in] normalTraction the normal traction
+   * @param[out] dLimitTangentialTractionNorm_dTraction the derivative of the limit tangential traction norm wrt normal traction
+   * @return the limit tangential traction norm
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  virtual real64 computeLimitTangentialTractionNorm( real64 const & normalTraction,
+                                                     real64 & dLimitTangentialTractionNorm_dTraction ) const
```

Comment:
```suggestion
  virtual real64 computeLimitTangentialTractionNorm( real64 const & normalTraction,
                                                     real64 & dLimitTangentialTractionNorm_dTraction )
```

Reply:
this method needs to be called from a `const` lambda. I think it must be marked `const`. Plus, it does seem const to me.

Reply:
I agree with @CusiniM, I left the `const` here